.github/workflows/lock-threads.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Lock Threads'
+
+on:
+  schedule:
+    - cron: '0 * * * *'
+  workflow_dispatch:
+
+permissions:
+  issues: write
+  pull-requests: write
+  discussions: write
+
+concurrency:
+  group: lock-threads
+
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dessant/lock-threads@v5
+        with:
+          issue-inactive-days: 30
+          issue-comment: >
+            This issue has been automatically locked since there
+            has not been any recent activity after it was closed.
+            Please open a new issue for related bugs.
+          pr-comment: >
```

Comment:
I wouldn't set this for PRs right now, some old PRs are good just abandoned and need some TLC)

Reply:
I think this is fine -- this is not an auto-close, but an auto-lock. If there is a good old PR that is closed and unmerged, someone should carry the changes as a new PR anyway.